### PR TITLE
Check HPOS feauture before quering DB for labels report

### PR DESCRIPTION
**Pull request by samnajian**
## Description
This PR adds support for HPOS in shipping labels report.
It also fixes a bug that prevented the query to include the whole days on the borders of the query range.

### Related issue(s)
https://github.com/woocommerce/woocommerce-shipping/issues/923

### Steps to reproduce & screenshots/GIFs
- With trunk of this repo, make a label purchase while your [test shop is using HPOS](https://woocommerce.com/document/high-performance-order-storage/#how-to-enable-high-performance-order-storage).
- Visit {site-domain}/wp-admin/admin.php?page=wc-reports&tab=wcs_labels and verify you can't see the recently purchase label in the report table.
- Now build and use this branch, visit the report page and verify the recently purchase label is viewable on the report table.

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added
- [x] `readme.txt` entry added

### CODE REVIEW

**Review by iyut — COMMENTED** [review]
@samnajian did you get this when you test the report? 
Screenshot 2024-11-15 at 21 28 33

I think this is unrelated to this PR though. But one idea that I can think of to fix this is by making sure the timestamp is int using `intval()` on [this line](https://github.com/Automattic/woocommerce-services/blob/0e30566118aa896604ef66aa3d4490bb29b7309e/classes/class-wc-connect-label-reports.php#L200) : `echo esc_html( get_date_from_gmt( date( 'Y-m-d H:i:s', intval( $label['created'] / 1000 ) ) ) );` WDYT?

File: classes/class-wc-connect-label-reports.php [hunk 1]
```diff
@@ -72,9 +79,15 @@ private function query_labels() {
...
-			// translate timestamps to JS timestapms
-			$start_date = $this->start_date * 1000;
-			$end_date   = $this->end_date * 1000;
+			/**
+			 * Translate timestamps to JS timestamps.
+			 *
+			 * The start_date is set to the beginning of the day (midnight) of the start_date property, converted to milliseconds.
+			 * The end_date is set to the end of the day (one millisecond before midnight) of the end_date property, converted to milliseconds.
+			 * This ensures that the date range includes the entire days specified by start_date and end_date.
+			 */
```
Comment (iyut) [inline classes/class-wc-connect-label-reports.php:88]: 👍 Thanks a lot for adding the explanation here.

**Review by iyut — APPROVED** [review]
The changes does fix the issue. LGTM! I just added a comment about notice error that I get. It's probably not related to this PR though.

**Comment by samnajian** [comment]
> I think this is unrelated to this PR though. But one idea that I can think of to fix this is by making sure the timestamp is int using `intval()` on [this line](https://github.com/Automattic/woocommerce-services/blob/0e30566118aa896604ef66aa3d4490bb29b7309e/classes/class-wc-connect-label-reports.php#L200) : `echo esc_html( get_date_from_gmt( date( 'Y-m-d H:i:s', intval( $label['created'] / 1000 ) ) ) );` WDYT?

@iyut No I didn't, how come the dates you have have decimal? basically it indicates miliseconds, didn't know we aim for that high level of precision, you're suggesting a nice idea of using `intval` around it.

**Comment by samnajian** [comment]
> I think this is unrelated to this PR though. But one idea that I can think of to fix this is by making sure the timestamp is int using `intval()` on [this line](https://github.com/Automattic/woocommerce-services/blob/0e30566118aa896604ef66aa3d4490bb29b7309e/classes/class-wc-connect-label-reports.php#L200) : `echo esc_html( get_date_from_gmt( date( 'Y-m-d H:i:s', intval( $label['created'] / 1000 ) ) ) );` WDYT?

Addressed in a15b67322c24ef93f0679713c1f1a019658b360a

**Pull request merged**